### PR TITLE
Sdf scaling

### PR DESCRIPTION
**Pull request by dbenson24**
This change adds the ability to scale the area the sdf texture represents and then uses bilinear interpolation in the shaders that read the sdf data. This gives an easily adjustable way to address offscreen shadow issues

Before Change
image

After Change
image

### CODE REVIEW

**Comment by dbenson24** [comment]
Before 
image

After
image

**Comment by dbenson24** [comment]
I also am seeing a performance boost with this change.
Before
```
2022-12-30T01:33:50.445300Z  INFO bevy diagnostic: frame_count                     : 689.000000
2022-12-30T01:33:51.444401Z  INFO bevy diagnostic: frame_time                      :   19.929964ms (avg 20.097590ms)
2022-12-30T01:33:51.444733Z  INFO bevy diagnostic: fps                             :   50.280973   (avg 49.885925)
```

After
```
2022-12-30T01:32:25.658418Z  INFO bevy diagnostic: frame_count                     : 1408.000000
2022-12-30T01:32:26.653097Z  INFO bevy diagnostic: frame_time                      :   18.884080ms (avg 18.671375ms)
2022-12-30T01:32:26.653402Z  INFO bevy diagnostic: fps                             :   53.029591   (avg 53.649381)
```

File: assets/shaders/gi_sdf.wgsl [hunk 1]
```diff
@@ -27,13 +27,14 @@ fn round_merge(s1: f32, s2: f32, r: f32) -> f32 {
...
 @compute @workgroup_size(8, 8, 1)
 fn main(@builtin(global_invocation_id) invocation_id: vec3<u32>) {
-     let screen_pose  = vec2<i32>(invocation_id.xy);
-     let world_pose   = screen_to_world(
-        screen_pose,
-        camera_params.screen_size,
+     let texel_pos  = vec2<i32>(invocation_id.xy);
+     let dims = textureDimensions(sdf_out);
+     let uv = (vec2<f32>(texel_pos) + 0.5) / vec2<f32>(dims);
+    //  let uv = vec2<f32>(uv.x, 1.0 - uv.y);
```
Comment (zaycev) [inline assets/shaders/gi_sdf.wgsl:33]: Let's remove commented-out stuff when possible.

File: assets/shaders/gi_ss_blend.wgsl [hunk 2]
```diff
@@ -7,18 +7,19 @@
 @group(0) @binding(0) var<uniform> camera_params:     CameraParams;
 @group(0) @binding(1) var<uniform> state:             GiState;
 @group(0) @binding(2) var<storage> probes:            ProbeDataBuffer;
-@group(0) @binding(3) var          sdf_in:            texture_storage_2d<r16float,    read>;
-@group(0) @binding(4) var          ss_bounce_in:      texture_storage_2d<rgba32float, read>;
-@group(0) @binding(5) var          ss_blend_out:      texture_storage_2d<rgba32float, write>;
+@group(0) @binding(3) var          sdf_in:            texture_2d<f32>;
```
Comment (zaycev) [inline assets/shaders/gi_ss_blend.wgsl:10]: Thanks! This makes more sense.

File: assets/shaders/gi_ss_blend.wgsl [hunk 3]
```diff
@@ -7,18 +7,19 @@
...
 
 fn distance_squared(a: vec2<f32>, b: vec2<f32>) -> f32 {
     let c = a - b;
     return dot(c, c);
 }
 
-fn get_sdf_screen(screen_pose: vec2<i32>) -> f32 {
-    return textureLoad(sdf_in, screen_pose).r;
-}
+// fn get_sdf_screen(screen_pose: vec2<i32>) -> f32 {
```
Comment (zaycev) [inline assets/shaders/gi_ss_blend.wgsl]: Same as above, let's just remove this.

File: assets/shaders/gi_ss_probe.wgsl [hunk 4]
```diff
@@ -22,15 +23,9 @@ fn distance_squared(a: vec2<f32>, b: vec2<f32>) -> f32 {
...
-fn get_sdf_screen(screen_pose: vec2<i32>) -> f32 {
-    return textureLoad(sdf_in, screen_pose).r;
-}
-
-fn get_sdf_world(world_pose: vec2<f32>) -> f32 {
-    let ndc = vec4<f32>(world_pose, 0.0, 1.0) * camera_params.view_proj;
-    let screen_pose = ndc_to_screen(ndc.xy, camera_params.screen_size);
-    return get_sdf_screen(screen_pose);
-}
+// fn get_sdf_screen(screen_pose: vec2<i32>) -> f32 {
```
Comment (zaycev) [inline assets/shaders/gi_ss_probe.wgsl:26]: Same as above.

File: src/gi/shaders/gi_camera.wgsl [hunk 5]
```diff
@@ -44,3 +46,35 @@ fn world_to_screen(
...
+
+fn sdf_uv_to_world(uv: vec2<f32>, inverse_view_proj: mat4x4<f32>, sdf_scale: vec2<f32>) -> vec2<f32> {
+    let y = 1.0 - uv.y;
+    let uv = vec2<f32>(uv.x, y);
+    let ndc_sdf = (uv * 2.0) - 1.0;
+    let ndc = ndc_sdf * sdf_scale;
+    return (inverse_view_proj * vec4<f32>(ndc, 0.0, 1.0)).xy;
+}
+
+fn bilinearSample(channel: i32, t: texture_2d<f32>, s: sampler, uv: vec2<f32>) -> f32 {
```
Comment (zaycev) [inline src/gi/shaders/gi_camera.wgsl]: nit. `bilinearSample` -> `bilinear_sample`

**Review by zaycev — APPROVED** [review]
Minor feedback regarding naming and commented code, otherwise LGTM.

**Pull request merged**